src/main/java/treebot/TreeBot.java

```diff
@@ -0,0 +1,49 @@
+package treebot;
+
+import commands.Command;
+import exception.TreeBotException;
+import tasks.TaskFactory;
+import tasks.TaskList;
+import utils.Parser;
+import utils.Storage;
+import utils.Ui;
+
+import java.io.FileNotFoundException;
+
+
+public class TreeBot {
+    private static final String EXIT_TOKEN = "bye";
+    private TaskList taskList;
+    private Ui ui = new Ui();
+    private Storage storage;
+    private Parser parser = new Parser(new TaskFactory());
+
+    public TreeBot(String filePath) {
+        storage = new Storage(filePath);
+        try {
+            taskList = new TaskList(storage.loadTasks());
+        } catch (FileNotFoundException e) {
+            taskList = new TaskList();
+        }
+
+    }
+
+    public void run() {
+        ui.showWelcome();
+
+        while (true) {
+            try {
+                String fullCommand = ui.readCommand();
+                Command c = parser.parse(fullCommand);
+
+                if (c.equals(EXIT_TOKEN)) {
+                    break;
+                }
```

Comment:
The exit process is implemented incorrectly - the user input should be compared directly against `EXIT_TOKEN`.

```suggestion
                if (fullCommand.equals(EXIT_TOKEN)) {
                    break;
                }

                Command c = parser.parse(fullCommand);
```